packages/plugin-trpc/cli/src/lib/generate.ts

```diff
@@ -0,0 +1,331 @@
+										.write(',')
+										.newLine();
+								});
+						})
+						.write(')');
+				}
+			}
+		]
+	});
+
+	if (nonScalarFileds.length > 0) {
+		nonScalarFileds.forEach((field) => {
+			if (field.type !== 'field') return;
+			const isEnum = determineEnum(field);
+			if (isEnum && isEnum.type === 'enum') {
+				sourceFile.addImportDeclaration({
+					moduleSpecifier: `../enums/${generateEnumFilename(isEnum.name)}`,
+					namedImports: [generateZodEnumName(isEnum.name), generateEnumName(isEnum.name)]
+				});
+			}
+			if (!isEnum) {
+				sourceFile.addImportDeclarations([
+					{
+						moduleSpecifier: `../models/${generateModelFilename(field.fieldType as string)}`,
+						namedImports: [`Complete${field.fieldType}Model`]
+					},
+					{
+						moduleSpecifier: `../models/${generateModelFilename(field.fieldType as string)}`,
+						namedImports: [`${field.fieldType as string}WithRelations`],
+						isTypeOnly: true
+					}
+				]);
+			}
+		});
+		sourceFile.addInterface({
+			isExported: true,
+			name: `${model.name}WithRelations`,
+			leadingTrivia: (writer) => writer.blankLineIfLastNot(),
+			trailingTrivia: (writer) => writer.blankLineIfLastNot(),
+			extends: [`z.infer<typeof ${generateZodModelName(model.name)}>`],
+			properties: nonScalarFileds.map((field) => {
+				let type = '';
+				if (field.type !== 'field') return { name: '', type: 'any' };
+				const isEnum = determineEnum(field);
+				if (isEnum && isEnum.type === 'enum') {
+					type = `${generateEnumName(isEnum.name)}${field.array ? '[]' : ''}${
+						field.optional ? ' | null' : ''
+					}`;
+				} else {
+					type = `${field.fieldType as string}WithRelations${field.array ? '[]' : ''}${
+						field.optional ? ' | null' : ''
+					}`;
+				}
+				return {
+					hasQuestionToken: field.optional,
+					name: field.name,
+					type
+				};
+			})
+		});
+
+		sourceFile.addVariableStatement({
+			declarationKind: VariableDeclarationKind.Const,
+			isExported: true,
+			leadingTrivia: (writer) => writer.blankLineIfLastNot(),
+			trailingTrivia: (writer) => writer.blankLineIfLastNot(),
+			declarations: [
+				{
+					name: `Complete${model.name}Model`,
+					type: `z.ZodSchema<${model.name}WithRelations>`,
+					initializer: (writer) => {
+						writer
+							.write(`z.lazy(() => ${generateZodModelName(model.name)}.extend(`)
+							.inlineBlock(() => {
+								nonScalarFileds.forEach((prop) => {
+									if (prop.type !== 'field') return;
+									writer
+										.write(`${prop.name}: ${mapZodType(prop)}`)
+										.write(',')
+										.newLine();
+								});
+							})
+							.write('))');
+					}
+				}
+			]
+		});
+	}
+};
+
+export const generateEnumSchema = (sourceFile: SourceFile, enumModel: Enum) => {
+	sourceFile.addImportDeclaration({
+		moduleSpecifier: 'webstone-plugin-web-trpc',
+		namedImports: ['z']
+	});
+
+	const enumValues = enumModel.enumerators.map(
+		(enumerator) => enumerator.type === 'enumerator' && enumerator.name
```

Comment:
I definitely want to create a Typeguard for the Prisma-types. We need to check very often, although we can very likely assume the type after checking once

Reply:
Edit: I'm not sure if this makes sense, because we still need to do that check before. We wouldn't get rid of the permanent checking, wouldn't we?